core/src/main/java/tc/oc/pgm/fireworks/FireworkMatchModule.java

```diff
@@ -0,0 +1,237 @@
+package tc.oc.pgm.fireworks;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import org.bukkit.Bukkit;
+import org.bukkit.Color;
+import org.bukkit.FireworkEffect;
+import org.bukkit.FireworkEffect.Type;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Firework;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.inventory.meta.FireworkMeta;
+import tc.oc.pgm.Config;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.match.event.MatchFinishEvent;
+import tc.oc.pgm.api.party.Competitor;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.controlpoint.events.ControllerChangeEvent;
+import tc.oc.pgm.core.CoreLeakEvent;
+import tc.oc.pgm.destroyable.DestroyableDestroyedEvent;
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.pgm.flag.event.FlagCaptureEvent;
+import tc.oc.pgm.regions.Bounds;
+import tc.oc.pgm.regions.Region;
+import tc.oc.pgm.wool.PlayerWoolPlaceEvent;
+import tc.oc.util.bukkit.BukkitUtils;
+import tc.oc.util.bukkit.block.BlockVectors;
+
+@ListenerScope(MatchScope.LOADED)
+public class FireworkMatchModule implements MatchModule, Listener {
+
+  private final Match match;
+
+  private final boolean postEnabled = Config.Fireworks.postMatchEnabled();
+  private final boolean goalsEnabled = Config.Fireworks.goalsEnabled();
+  private final int number = 5;
+  private final int delay = 40;
+  private final int frequency = 40;
+  private final int iterationCount = 15;
+  private final int power = 2;
```

Comment:
Should be static, and follow the static variable naming rules: `ITERATION_COUNT`